crates/hyperswitch_connectors/src/connectors/nexixpay/transformers.rs

```diff
@@ -1,80 +1,405 @@
+    operation_id: String,
+    capture_type: Option<NexixpayCaptureType>,
+    three_d_s_auth_data: ThreeDSAuthData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OperationData {
+    operation_id: String,
+    operation_currency: Currency,
+    operation_result: NexixpayPaymentStatus,
+    operation_type: NexixpayOperationType,
+    order_id: String,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayCompleteAuthorizeResponse {
+    operation: OperationData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPreProcessingRequest {
+    operation_id: String,
+    three_d_s_auth_response: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Order {
+    order_id: String,
+    amount: StringMinorUnit,
+    currency: Currency,
+    description: Option<String>,
+    customer_info: CustomerInfo,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CustomerInfo {
+    card_holder_name: Secret<String>,
+    billing_address: Address,
+    shipping_address: Option<Address>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Address {
+    name: Secret<String>,
+    street: Secret<String>,
+    city: String,
+    post_code: Secret<String>,
+    country: enums::CountryAlpha2,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct NexixpayCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+    pan: CardNumber,
+    expiry_date: Secret<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Recurrence {
+    action: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayPaymentsResponse {
+    operation: Operation,
+    three_d_s_auth_request: String,
+    three_d_s_auth_url: Secret<url::Url>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ThreeDSAuthResult {
+    authentication_value: Secret<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub enum NexixpayPaymentIntent {
+    Capture,
+    Cancel,
+    Authorize,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NexixpayConnectorMetaData {
+    pub three_d_s_auth_result: Option<ThreeDSAuthResult>,
+    pub three_d_s_auth_response: Option<String>,
+    pub authorization_operation_id: Option<String>,
+    pub capture_operation_id: Option<String>,
+    pub cancel_operation_id: Option<String>,
+    pub psync_flow: NexixpayPaymentIntent,
```

Comment:
`psync_flow` might be an unnecessary field. Please explain the usecase.
I think it can be inferred from the request itself

Reply:
As nexi pay returns new payment id in capture and cancel calls and for doing psync call we should use these ids. Thus in geturl function of psync we consume this field to check which payment id we should use.